pkg/chunk/cassandra/storage_client.go

```diff
@@ -49,6 +51,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.Auth, "cassandra.auth", false, "Enable password authentication when connecting to cassandra.")
 	f.StringVar(&cfg.Username, "cassandra.username", "", "Username to use when connecting to cassandra.")
 	f.StringVar(&cfg.Password, "cassandra.password", "", "Password to use when connecting to cassandra.")
+	f.StringVar(&cfg.PasswordFile, "cassandra.password-file", "", "File containing password to use when connecting to cassandra.")
```

Comment:
I would suggest to add a `Config.Validate()` function to validate this config and make sure both `Password` and `PasswordFile` are not set (they should be mutually exclusive). Then you should call the `Validate()` from the chain of config back to `cortex.Config.Validate()`  (see how other `Validate()`  functions are implemented, please).